src/relay/pass/type_solver.h

```diff
@@ -40,13 +42,13 @@ using common::LinkedList;
  */
 class TypeSolver {
  public:
-  TypeSolver();
+  TypeSolver(const GlobalVar& current_func, ErrorReporter* err_reporter);
   ~TypeSolver();
   /*!
    * \brief Add a type constraint to the solver.
    * \param constraint The constraint to be added.
    */
-  void AddConstraint(const TypeConstraint& constraint);
+  void AddConstraint(const TypeConstraint& constraint, const NodeRef& lcoation);
```

Comment:
s/lcoation/location, and document it if necessary.  Should we pass `location` to these two methods? It looks a little strange to me.

Reply:
I still have some refactoring to do after talking with Tianqi on Friday. We were trying to settle on a design. The motivation is that a constraint arises from a `location` i.e a `Expr` or `Type` and the error should be reported.